### PR TITLE
rvalue makeMeta returns value

### PR DESCRIPTION
**Pull request by cryogenian**

### CODE REVIEW

**Comment by cryogenian** [comment]
[ch3721]

**Comment by djspiewak** [comment]
I think the correct approach to solving this is actually just adding support for metadata. It should mirror the yggdrasil encoding, rather than the ejson encoding.

**Comment by djspiewak** [comment]
```scala
final case class RMeta(value: RValue, namespace: RObject) extends RValue
```

**Comment by cryogenian** [comment]
@djspiewak 

I've added `RMeta` but I'm not totally sure what `RValue` is 🙁 idk if I need to add additional cases here 
https://github.com/slamdata/quasar/blob/5d34f6cf740db605c324a3d93ff1a30f792891fe/impl/src/main/scala/quasar/impl/evaluate/RValueParseInstructionInterpreter.scala#L63-L80

File: common/src/main/scala/quasar/common/data/QDataRValue.scala [hunk 1]
```diff
@@ -168,7 +169,16 @@ object QDataRValue extends QDataEncode[RValue] with QDataDecode[RValue] {
...
+    case _ => error(s"Expected `RMeta`. Received $a")
+  }
+  def getMetaMeta(a: RValue): RValue = a match {
+    case RMeta(_, meta) => meta
+    case _ => error(s"Expected `RMeta`. Received $a")
+  }
+  def makeMeta(value: RValue, meta: RValue): RValue = meta match {
+    case robj@RObject(obj) => RMeta(value, robj)
+    case _ => error(s"Expected `RObject`. Received $meta")
+  }
```
[the remark this answers is not in the document]
Comment (cryogenian) [reply common/src/main/scala/quasar/common/data/QDataRValue.scala]: Actually why is it necessary to wrap meta `RValue` into `RObject`? 
```scala
def makeMeta(value: RValue, meta: RValue): RValue = RMeta(value, meta)
```
?

**Comment by cryogenian** [comment]
+ `def makeMeta(value: RValue, meta: RValue): RValue = RMeta(value, meta)`

File: common/src/main/scala/quasar/common/data/RValue.scala [hunk 2]
```diff
@@ -324,7 +324,7 @@ sealed abstract class RValueInstances {
   implicit val qdataDecode: QDataDecode[RValue] = QDataRValue
 }
 
-final case class RMeta(value: RValue, meta: RObject) extends RValue
+final case class RMeta(value: RValue, meta: RValue) extends RValue
```
Comment (djspiewak) [inline common/src/main/scala/quasar/common/data/RValue.scala]: Nope. :-) Take a look at `Slice` and `CPath` (which is inside of `ColumnRef`). Note the `CPathMeta` case. `RValue` needs to be a 1-to-1 correspondence with *that* model.

**Comment by cryogenian** [comment]
@djspiewak Something like this?

**Comment by djspiewak** [comment]
@cryogenian Yes definitely! Also don't force-push please. :-) But that looks exactly like what it should be.

**Comment by cryogenian** [comment]
> Also don't force-push please. :-) 

:+1:

**Comment by beckyconning** [comment]
💯

**Pull request merged**